src/inference/forwardSample.js

```diff
@@ -68,6 +70,12 @@ module.exports = function(env) {
     },
 
     factor: function(s, k, a, score) {
+      if (!this.opts.guide && !this.factorWarningIssued) {
```

Comment:
Hmm.. I understand that use of a guide, but quite can't manage to see why that's a good reason to show a warning here. If you're happy to go ahead with this as it, we maybe resolve this some other time?

One thing that might be worth clarifying:

> Why don't we issue the warning if a guide is present?

The condition isn't whether there's a guide present, it's whether the `{guide: true}` argument is passed to `Infer`. (There's kinda *always* a guide present -- either the one written explicitly in the program, or the mean-field thing we give you otherwise.)